lib/cisco_node_utils/router_ospf_area.rb

```diff
@@ -0,0 +1,285 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ipaddr'
+require_relative 'node_util'
+require_relative 'router_ospf'
+require_relative 'router_ospf_vrf'
+
+module Cisco
+  # node_utils class for ospf_area
+  class RouterOspfArea < NodeUtil
+    attr_reader :router, :vrf, :area_id
+
+    def initialize(ospf_router, vrf_name, area_id)
+      fail TypeError unless ospf_router.is_a?(String)
+      fail TypeError unless vrf_name.is_a?(String)
+      fail TypeError unless area_id.is_a?(String)
+      fail ArgumentError unless ospf_router.length > 0
+      fail ArgumentError unless vrf_name.length > 0
+      fail ArgumentError unless area_id.length > 0
+
+      # check the area_id is a proper ipv4 address
+      fail ArgumentError if (begin
+                               IPAddr.new(area_id)
+                             rescue
+                               nil
+                             end).nil?
+      fail ArgumentError unless IPAddr.new(area_id).ipv4?
+
+      @router = ospf_router
+      @vrf = vrf_name
+      @area_id = area_id
+
+      set_args_keys_default
+    end
+
+    def self.areas
+      hash = {}
+      RouterOspf.routers.each do |instance|
+        name = instance[0]
+        # get all area ids under default vrf
+        area_ids = config_get('ospf_area', 'areas', name: name)
+        unless area_ids.nil?
+          hash[name] = {}
+          hash[name]['default'] = {}
+          area_ids.uniq.each do |area|
+            hash[name]['default'][area] =
+              RouterOspfArea.new(name, 'default', area)
+          end
+        end
+        vrf_ids = config_get('ospf', 'vrf', name: name)
+        next if vrf_ids.nil?
+        vrf_ids.each do |vrf|
+          # get all area ids under each vrf
+          area_ids = config_get('ospf_area', 'areas', name: name, vrf: vrf)
+          next if area_ids.nil?
+          hash[name] = {} if hash.empty?
+          hash[name][vrf] = {}
+          area_ids.uniq.each do |area|
+            hash[name][vrf][area] =
+              RouterOspfArea.new(name, vrf, area)
+          end
+        end
+      end
+      hash
+    end
+
+    # Helper method to delete @set_args hash keys
+    def set_args_keys_default
+      @set_args = { name: @router }
+      @set_args[:vrf] = @vrf unless @vrf == 'default'
+      @set_args[:area] = @area_id
+      @get_args = @set_args
+    end
+
+    # rubocop:disable Style/AccessorMethodName
+    def set_args_keys(hash={})
+      set_args_keys_default
+      @set_args = @get_args.merge!(hash) unless hash.empty?
+    end
+
+    def destroy
+      return unless Feature.ospf_enabled?
```

Comment:
This probably needs a second check... I'm thinking about the case where I've just created this object but I haven't set any properties yet, but then follow immediately with a destroy. The result on the device will be a be a router and vrf configuration where maybe there was none before.